### DISCUSSION TITLE
g2 如何给画布背景色，下载图片时背景色是透明的，使用theme无作用

### DISCUSSION THREAD

**Discussion by git-csy** [antvis/G2#5112] [Q&A]
* **G2 Version**:
* **Platform**:
* **Mini Showcase(like screenshots)**:
* **CodePen Link**:

**Comment by git-csy** [comment]
![企业微信截图_15900607623106](https://user-images.githubusercontent.com/52741749/82555179-e0ada780-9b99-11ea-93c2-5fcb6a67891d.png)
主题还是存在部分问题吧

**Comment by Sai0514** [comment]
同问，并且theme目前只有‘dark’生效，background也是加在div上面，似乎并不是绘制在canvas上面的？

**Comment by hustcc** [comment]
4.1已经改成不用 css 了，可以导出黑色背景了。

**Comment by Arco1ris** [comment]
如果渲染的时候配置了 slider，下载的时候，slider 及其周边还是透明的（修改 slider 背景色也只改变了 slider 本身的背景色，它的周边还是透明色），只有上面坐标轴区域是 theme 配置的背景色。截图是下载后的图片
<img width="1118" alt="截屏2022-06-22 下午3 02 03" src="https://user-images.githubusercontent.com/10501603/174964859-a6cd903d-fac4-43a1-a876-12d2d5f29006.png">
。

**Comment by Arco1ris** [comment]
@hustcc 请问这种情况应如何处理

**Comment by hustcc** [comment]
slider 没有使用 theme 中的背景色是这个意思嘛？

**Comment by Arco1ris** [comment]
> slider 没有使用 theme 中的背景色是这个意思嘛？

是的

**Comment by hustcc** [comment]
主题中的 background 是印象 view，理论上 view 的大小是包含 slider 区域的，这个我们排查下。 ref https://github.com/antvis/G2/issues/3299

**Comment by hustcc** [comment]
尝试两个 PR 修复，都修复不了，设计问题导致修复会引入其他问题。这个问题在 G2 5.0 中在设计的时候考虑进去。见 [DEMO](https://g2.antv.antgroup.com/examples/style/theme/#layout-area)。
